main/commons/src/main/java/org/cryptomator/common/mountpoint/MountPointChooserModule.java

```diff
@@ -0,0 +1,59 @@
+package org.cryptomator.common.mountpoint;
+
+import com.google.common.collect.ImmutableSet;
+import dagger.Module;
+import dagger.Provides;
+import dagger.multibindings.IntoSet;
+import org.cryptomator.common.Environment;
+import org.cryptomator.common.vaults.PerVault;
+import org.cryptomator.common.vaults.Vault;
+import org.cryptomator.common.vaults.WindowsDriveLetters;
+
+import javax.inject.Named;
+import java.util.Set;
+
+/**
+ * Dagger-Module for {@link MountPointChooser MountPointChoosers.}<br>
+ * See there for additional information.
+ *
+ * @see MountPointChooser
+ */
+@Module
+public abstract class MountPointChooserModule {
+
+	@Provides
+	@IntoSet
+	@PerVault
+	public static MountPointChooser provideCustomMountPointChooser(Vault vault) {
+		return new CustomMountPointChooser(vault);
+	}
+
+	@Provides
+	@IntoSet
+	@PerVault
+	public static MountPointChooser provideCustomDriveLetterChooser(Vault vault) {
+		return new CustomDriveLetterChooser(vault);
+	}
+
+	@Provides
+	@IntoSet
+	@PerVault
+	public static MountPointChooser provideAvailableDriveLetterChooser(WindowsDriveLetters windowsDriveLetters) {
+		return new AvailableDriveLetterChooser(windowsDriveLetters);
+	}
+
+	@Provides
+	@IntoSet
+	@PerVault
+	public static MountPointChooser provideTemporaryMountPointChooser(Vault vault, Environment environment) {
+		return new TemporaryMountPointChooser(vault, environment);
+	}
+
+	@Provides
+	@PerVault
+	@Named("orderedValidMountPointChoosers")
+	public static Set<MountPointChooser> provideOrderedValidMountPointChoosers(Set<MountPointChooser> choosers) {
+		//Sorted Set
+		return choosers.stream().sorted().filter(MountPointChooser::isApplicable).collect(ImmutableSet.toImmutableSet());
+	}
```

Comment:
Collect using `collect(Collectors.toList())`.

Even if the current implementation of `toImmutableSet()` was order-preserving (which I doubt), this might change any time, as long as it isn't part of the contract.

Reply:
> `Collectors#toList()`'s docs give no clue about whether is order-preserving or not. Also I was looking for a Set, not a List.

Well it says `toList` and returns a `List`. It is an inherent properties of Lists to preserve the order of Elements.

Why is it you want a Set?

Reply:
>Well it says toList and returns a List. It is an inherent properties of Lists to preserve the order of Elements.

Sorry, let me rephrase: The docs give no exact definition of which order the list will have, even if it's usually the encounter order of the adding stream. 

>Why is it you want a Set?

Sets don't allow double elements, which reflects the nature of the MPCs as "PerVault"-Singletons. Also `@IntoSet` generates a Set and building a List from a Set seems strange to me.

Reply:
I don't think you got the point. Yes, the param `choosers` it is a set with unique elements, collected using `@IntoSet`.

However, I am talking about the **returned** collection. In this case, you provide a collection of elements **deliberately sorted** by some criteria. If this order is relevant to whoever wants to use this collection (which it is when iterating over elements in order of preference) you want to signify it using an appropriate collection type.

In case of doubt, someone who'd be injecting this collection without knowing the implementation of this module has _no clue_ that the Set was in fact a SortedSet.

Reply:
> In case of doubt, someone who'd be injecting this collection without knowing the implementation of this module has no clue that the Set was in fact a SortedSet.

The name of the set `@Named("orderedValidMountPointChoosers")` should usually make that clear, but I understand, what you mean. 

I would like to keep using a child interface of Set to reflect this nature of the collection:
> I don't think you got the point. Yes, the param choosers it is a set with unique elements, collected using @IntoSet.

> However, I am talking about the returned collection. In this case, you provide a collection of elements deliberately sorted by some criteria. If this order is relevant to whoever wants to use this collection (which it is when iterating over elements in order of preference) **you want to signify it using an appropriate collection type.**

To make things clearer, I will change the method to use `ImmutableSortedSet's Collector` and return a `SortedSet.`

Reply:
9657a13912d0e68f68f54235dd68cceb7a0c93e0 should hopefully fix that issue.

Reply:
It fixes it, indeed.